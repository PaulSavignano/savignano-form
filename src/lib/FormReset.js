import React from 'react'
import PropTypes from 'prop-types'

import Context from './Context'

function FormReset(props) {
  const { component: Comp, names, ...rest } = props
  return (
    <Context.Consumer>
      {({ onReset }) => (
        <Comp
          {...rest}
          onClick={() => onReset(names)}
          type="button"
        />
      )}
    </Context.Consumer>
  )
}

FormReset.defaultProps = {
  names: []
}

FormReset.propTypes = {
  component: PropTypes.func.isRequired,
  names: PropTypes.arrayOf(PropTypes.string)
}

export default FormReset
