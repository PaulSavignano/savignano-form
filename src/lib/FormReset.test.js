import React from 'react'
import { shallow } from 'enzyme'

import FormReset from './FormReset'

const testProps = {
  children: jest.fn(),
  component: p => <button {...p} type="button" />
}

const testCtx = {
  onReset: jest.fn()
}

describe('FormReset', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<FormReset {...testProps} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render child component', () => {
    const wrapper = shallow(<FormReset {...testProps} />)
    const contextWrapper = shallow(wrapper.prop('children')({ ...testCtx }))
    expect(contextWrapper.find('button').exists()).toEqual(true)
  })
  it('should call onReset when clicked', () => {
    const wrapper = shallow(<FormReset {...testProps} />)
    const spy = jest.fn()
    const contextWrapper = shallow(wrapper.prop('children')({ onReset: spy }))
    const button = contextWrapper.find('button')
    button.props().onClick({ preventDefault: jest.fn() })
    expect(spy).toHaveBeenCalled()
  })
})
