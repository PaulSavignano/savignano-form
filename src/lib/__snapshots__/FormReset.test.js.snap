// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormReset should match snapshot 1`] = `ShallowWrapper {}`;
